### PR TITLE
FLAG-77: Fix unit test execution issue during maven build

### PR DESCRIPTION
**Pull request by ManojLL**
Issue worked on : [FLAG-77](https://openmrs.atlassian.net/browse/FLAG-77)

[FLAG-77]: https://openmrs.atlassian.net/browse/FLAG-77?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -150,6 +156,14 @@
 					<artifactId>maven-openmrs-plugin</artifactId>
 					<version>1.0.1</version>
 				</plugin>
+				<plugin>
+					<groupId>org.apache.maven.plugins</groupId>
+					<artifactId>maven-surefire-plugin</artifactId>
+					<version>2.22.1</version>
+					<configuration>
+						<argLine>-Xmx512m -Djdk.net.URLClassPath.disableClassPathURLCheck=true</argLine>
```
Comment (wikumChamith) [inline pom.xml:164]: Do we need this argLine?

File: pom.xml [hunk 2]
```diff
@@ -204,5 +218,24 @@
...
+			<id>Java 17</id>
+			<activation>
+				<jdk>17</jdk>
+			</activation>
+			<build>
+				<plugins>
+					<plugin>
+						<groupId>org.apache.maven.plugins</groupId>
+						<artifactId>maven-surefire-plugin</artifactId>
+						<version>3.2.2</version>
```
Comment (wikumChamith) [inline pom.xml:231]: Can't we use the same version as Java 8?

File: pom.xml (hunk 1 above)
Comment (ManojLL) [reply pom.xml:164]: no, we don't need this argLine, I added this to fix the testNG suit issue, but the issue is still there.

File: pom.xml (hunk 2 above)
Comment (ManojLL) [reply pom.xml]: we use the same version as used in java 8

Comment (wikumChamith) [reply pom.xml]: Just so you know, you don't need to state the version here explicitly.  You could remove the version here.

**Comment by wikumChamith** [comment]
Do the tests work for you now?

**Comment by ManojLL** [comment]
> Do the tests work for you now?

Tests are executing but stills having test suit issue
image

**Comment by wikumChamith** [comment]
@ManojLL Can we rename the PR name to reflect what we are achieving from this PR?

File: pom.xml [hunk 3]
```diff
@@ -204,5 +210,24 @@
 			</properties>
 		</profile>
 
+		<profile>
```
Comment (wikumChamith) [inline pom.xml:213]: It would be great if we could get this as a different PR.

File: pom.xml [hunk 4]
```diff
@@ -119,6 +119,12 @@
 				<version>${legacyuiVersion}</version>
 				<scope>provided</scope>
 			</dependency>
+			<dependency>
```
Comment (wikumChamith) [inline pom.xml:122]: Do we need this now?

File: pom.xml (hunk 3 above)
Comment (ManojLL) [reply pom.xml:213]: If we remove this profile section, project build for Java 11 and Java 17 will fail for this PR

File: pom.xml (hunk 4 above)
Comment (ManojLL) [reply pom.xml:122]: yes, this decency is needed for java 17

File: pom.xml (hunk 3 above)
Comment (wikumChamith) [reply pom.xml:213]: We could rebase that commit to this pull request. So that won't be a problem.

File: pom.xml (hunk 4 above)
Comment (wikumChamith) [reply pom.xml:122]: You could also add this to the Java 17 pull request.

File: pom.xml (hunk 3 above)
Comment (wikumChamith) [reply pom.xml]: @ManojLL I merged the pull request for Java 17 support.

Comment (ManojLL) [reply pom.xml]: I debase the branch with java 17 changes

**Pull request merged**